src/Analyzer/Passes/SumIfToCountIfPass.cpp

```diff
@@ -32,7 +32,7 @@ class SumIfToCountIfVisitor : public InDepthQueryTreeVisitorWithContext<SumIfToC
             return;
 
         auto * function_node = node->as<FunctionNode>();
-        if (!function_node || !function_node->isAggregateFunction())
+        if (!function_node || !function_node->isAggregateFunction() || function_node->getResultType()->isNullable())
```

Comment:
This reminds me of https://github.com/ClickHouse/ClickHouse/pull/60453/commits/f0354ee07f4e0641c77b92a82711b6efb508fcb2.

Both bugs are introduced because `getValue()` ignores nullability.

Probably `constant_value_literal.getType()` should be changed to `literal->getResultType()`, and `getType()` should be refactored to something with a completely different name to avoid this confusion again as `getResultType()` and `getType()` and pretty much identical and `getType()` does unexpected things. Thoughts?

Reply:
yes it was confusing at first when reading the code because you don't pay attention to the fact that the Nullability info is lost when you get Field
not sure what would be the best solution